src/app/panels/image-panel/image-panel.component.ts

```diff
@@ -22,9 +36,16 @@ export class ImagePanelComponent {
 
   constructor(
     private evtModelService: EVTModelService,
+    private evtStatus: EVTStatusService,
   ) {
   }
 
+  updatePage(viewerPage: number){
+    this.evtModelService.pages$.pipe(take(1)).subscribe(
```

Comment:
Create a new subscription at every page change creates a memory leak.
Ensure there is a single subscription (e.g. inside the constructor) and chain the event stream of the osd-component to an observable property of this component.

Reply:
@szenzaro Can you suggest the better way of doing it?

Reply:
Check and review last commit